### ISSUE TITLE
Make cubos compile faster

### ISSUE THREAD

**Issue by RiscadoA** [GameDevTecnico/cubos#365]

**Comment by RiscadoA** [comment]
Our biggest issue is the huge linking times - using dynamic linking for the engine & core might solve this problem.

**Comment by roby2014** [comment]
Should we write some sort of blog post / information on docs on how we can reduce compile time using some tricks? (e.g, use glfw and glm externally, ccache, ....(hopefully new tricks in the future)?

**Comment by RiscadoA** [comment]
Yeah that definitely would a be a good idea. I think it would fit better in the docs since the blog posts are more community facing.

**Comment by roby2014** [comment]
Yeah I can write something after we find out what the hell is happening here: #834
